nx_main/loaders/builtin.c

```diff
@@ -45,16 +46,30 @@ static bool init(void)
 
 static void deinit(void)
 {
-    
+
 }
 
 static void launchFile(const char* path, argData_s* args)
 {
     /*if (strncmp(path, "sdmc:/",6) == 0)
         path += 5;*/
     memset(argBuf, 0, sizeof(argBuf));
+
+   uint32_t remote = args->nxlink_host.s_addr;
+
+   if (remote) {
+
+    char nxlinked[17] = "NXLINKED";
+
+        itoa(remote,&nxlinked[8],16);
+
+        launchAddArg(args, nxlinked);
```

Comment:
Maybe instead of adding a real argv arg, we can replace argv[0] with NXLINKED or something? Thus we won't intrude on argv support (what if someone has a file name that's NXLINKED and puts that as arg, it will incorrectly be interpreted as nxlink presence)

Reply:
Additional arg doesn't matter *if* the app uses latest libnx. The actual arg is "NXLINKED{ipaddr}".

Reply:
I've changed this from itoa to sprintf so we can guarantee length of the argument & added a check to libnx for the arg length. "NXLINKED" is just a tag, we can change this to "_NXLINK_" if you think it's better but I don't really see it being an issue since it's quite specific already.

Reply:
OK, changed this around so it's {ipaddr}_NXLINK_ so it's a pretty specific signature. If someone is passing an arg that matches that pattern then they're just taking the piss.

Reply:
I suggest `%08x` to ensure that the address always has 8 hex digits.